pyemma/msm/estimators/_OOM_MSM.py

```diff
@@ -0,0 +1,261 @@
+import numpy as np
+import scipy.linalg as scl
+import scipy.sparse
+from pyemma.util.linalg import _sort_by_norm
+import msmtools.estimation as me
+
+__all__ = ['bootstrapping_count_matrix', 'twostep_count_matrix', 'rank_decision',
+           'oom_components', 'equilibrium_transition_matrix']
+
+
+def bootstrapping_count_matrix(Ct, nbs=500):
+    """
+    Perform bootstrapping on trajectories to estimate uncertainties for singular values of count matrices.
+
+    Parameters
+    ----------
+    Ct : csr-matrix
+        count matrix of the data.
+
+    nbs : int, optional, default=1000
+        the number of re-samplings to be drawn from dtrajs
+
+    Returns
+    -------
+    smean : ndarray(N,)
+        mean values of singular values
+    sdev : ndarray(N,)
+        standard deviations of singular values
+    """
+    # Get the number of states:
+    N = Ct.shape[0]
+    # Get the number of transition pairs:
+    T = Ct.sum()
+    # Reshape and normalize the count matrix:
+    p = Ct.toarray()
+    p = np.reshape(p, (N*N,)).astype(np.float)
+    p = p / T
+    # Perform the bootstrapping:
+    svals = np.zeros((nbs, N))
+    for s in range(nbs):
+        # Draw sample:
+        sel = np.random.multinomial(T, p)
+        # Compute the count-matrix:
+        sC = np.reshape(sel, (N, N))
+        # Compute singular values:
+        svals[s, :] = scl.svdvals(sC)
+    # Compute mean and uncertainties:
+    smean = np.mean(svals, axis=0)
+    sdev = np.std(svals, axis=0)
+
+    return smean, sdev
+
+def twostep_count_matrix(dtrajs, lag, N):
```

Comment:
Is there a specific reason why you want the result in a 3-dimensional sparse array? If not, perhaps it's easier to use a list of N csr matrices, each being the 2-step matrix with (i, n, j). If yes, it seems more intuitive to use a csr matrix with N rows, each having a sparse vector of the up to NxN transitions observed in the (i, n, j) matrix.